kcwitools/extract_optimal.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import print_function
+
+import numpy as np
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from kcwitools import optimal_extract_utility as r
+from linetools.spectra.xspectrum1d import XSpectrum1D
+
+
+def extract_optimal_1D(fluxcube, varcube,header,wavegrid,x1,x2,y1,y2,outfile=None,verbose=False,plot=False):
+	"""
+
+	Args:
+		fluxcube: 3D kcwi fluxcube
+		varcube: 3D KCWI variance cube
+		header:  header information
+		wavegrid:  1D kcwi wavelength grid
+		
+		x1,x2,y1,y2: Four corners of an aribaty rectangular pseudo-slit for extraction
+		outfile: str, optional
+		  FITS file name
+		verbose [False] : optional, prints information on terminal
+		plot [False] : optional plots diagnostic stuff
+	
+	Returns:
+		xspec: XSpectrum1D
+
+	"""	
+
+	# CHECK CHECK CHECK
+	#THESE VALUES NEED TO BE DOUBLE CHECKED. WHICH ONE SHOULD I USE?
+	gain=header['GAIN2']
```

Comment:
As these may be useful elsewhere, I suggest we start a `defs.py`
module and put them in there.

Reply:
Done. Please add more stuff there.